src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SecureGroovyScriptTest.java

```diff
@@ -802,5 +802,40 @@ public void testSandboxClassResolution() throws Exception {
             assertTrue(e.getMessage().contains("staticMethod java.lang.System gc"));
         }
     }
-    
+
+    @Test public void testSandboxUsesSelectedBinding() throws Exception {
+        SecureGroovyScript sgs = new SecureGroovyScript("return a++", true, null);
+        Binding b = new Binding();
+        b.setVariable("a", 5);
+        SecureGroovyScript.PreparedScript script = sgs.configuringWithKeyItem().prepare(Jenkins.getInstance().getPluginManager().uberClassLoader, b);
+        try {
+            Object res = script.run();
+            assertTrue((int) res == 5);
+            assertTrue((int) b.getVariable("a") == 6);
+            res = script.run();
+            assertTrue((int) res == 6);
+            assertTrue((int) b.getVariable("a") == 7);
+        } catch (Exception e) {
+            script.cleanUp();
```

Comment:
This should be in a `finally` block, not a `catch` block (or just use `Closeable` trick mentioned above).

Reply:
strange that I did this, I'll check your `Closeable` suggestion